Dockerfile.build

```diff
@@ -1,2 +1,19 @@
 FROM golang:1.8.3
-RUN curl -s -o - https://get.docker.com/builds/Linux/x86_64/docker-1.9.1 > /usr/local/bin/docker && chmod +x /usr/local/bin/docker
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update -qq \
+ && apt-get install -qq -y --force-yes \
+     apt-transport-https \
+     apt-utils \
+     ca-certificates \
+     curl \
+     gnupg2 \
+     software-properties-common
+RUN curl -fsSL https://download.docker.com/linux/$(. /etc/os-release; echo "$ID")/gpg | apt-key add -
+RUN add-apt-repository \
+   "deb [arch=amd64] https://download.docker.com/linux/$(. /etc/os-release; echo "$ID") \
+   $(lsb_release -cs) \
+   stable"
+RUN apt-get update -qq \
+ && apt-get install -qq -y --force-yes docker-ce=17.12.0~ce-0~debian
```

Comment:
no need to suffix `~ce-0~debian` iirc

Reply:
```
E: Version '17.12.0' for 'docker-ce' was not found
```

Reply:
shite. the web lied.

Reply:
i generally test before i open PRs 😄

Reply:
Windows is a turd when it comes to development